### PR TITLE
Add Avalanche/Flatten to REPL

### PR DESCRIPTION
**Pull request by tranma**
* added options to show avalanche/flattened output
* make commands friendlier
* complete pretty output

### CODE REVIEW

File: src/Icicle/Data.hs [hunk 1]
```diff
@@ -76,29 +79,47 @@ data Value =
...
-    DoubleValue d  -> PP.double d
-    BooleanValue b -> PP.pretty b
-    -- I'm too lazy
-    _              -> PP.text (show v)
+    StringValue  t  -> PP.text (unpack t)
+    IntValue     i  -> PP.int i
+    DoubleValue  d  -> PP.double d
+    BooleanValue b  -> PP.pretty b
+    DateValue    d  -> PP.pretty d
+    StructValue  s  -> PP.pretty s
```
Comment (charleso) [inline src/Icicle/Data.hs]: Thanks. :)

Comment (tranma) [inline src/Icicle/Data.hs]: https://twitter.com/lambda_foo/status/611338239425646592
I felt guilty

Comment (tmcgilchrist) [inline src/Icicle/Data.hs]: heh, just something I feel rather strongly about. 
Didn't mean to offend.

Comment (michaelneale) [inline src/Icicle/Data.hs]: @tmcgilchrist you monster

Comment (amosr) [inline src/Icicle/Data.hs]: you should see all the "| otherwise" I have

File: src/Icicle/Repl.hs [hunk 2]
```diff
@@ -43,6 +42,7 @@ data ReplError
  | ReplErrorConvert (STC.ConvertError SP.SourcePos Var)
  | ReplErrorDecode  S.ParseError
  | ReplErrorRuntime S.SimulateError
+ | ReplErrorFlatten (AS.FlattenError Text)
```
Comment (amosr) [inline src/Icicle/Repl.hs]: oh no, but this doesn't have a case in Pretty?

**Comment by tranma** [comment]
self-approved :clap:

**Pull request merged**